src/storage/src/table/mod.rs

```diff
@@ -107,6 +111,29 @@ pub trait TableIter: Send {
     }
 }
 
+pub struct ExtractTableKeyIterator<I: StateStoreIter<Item = (FullKey<Vec<u8>>, Bytes)> + 'static> {
+    iter: I,
+}
+
+impl<I: StateStoreIter<Item = (FullKey<Vec<u8>>, Bytes)>> StateStoreIter
```

Comment:
Can we try implementing a `map` method for `StateStoreIter` by adding a trait named like `StateStoreIterMapExt`?